packages/ra-core/src/controller/edit/useEditController.ts

```diff
@@ -130,29 +130,49 @@ export const useEditController = <
                     : data
             ).then((data: Partial<RecordType>) => {
                 const mutate = getMutateWithMiddlewares(update);
+                let postEditArgument;
+
                 return mutate(
                     resource,
                     { id, data },
                     {
                         onSuccess: async (data, variables, context) => {
+                            if (context.snapshot[0][0][0] == 'posts') {
+                                postEditArgument =
+                                    data.title !== ' ' &&
+                                    data.teaser !== ' ' &&
+                                    data.title.trim().length !== 0 &&
+                                    data.teaser.trim().length !== 0;
+                            }
```

Comment:
How can you expect this code to work on any other page that the PostEdit page of the demo? :thinking: